src/vm/runners/cairo_runner.rs

```diff
@@ -895,13 +895,18 @@ impl CairoRunner {
         let segment_index = builtin.base();
         #[allow(deprecated)]
         for i in 0..segment_used_sizes[segment_index] {
-            let value = vm
+            let formatted_value = match vm
                 .segments
                 .memory
-                .get_integer((segment_index as isize, i).into())
-                .map_err(|_| RunnerError::MemoryGet((segment_index as isize, i).into()))?
-                .to_bigint();
-            writeln!(stdout, "{value}").map_err(|_| RunnerError::WriteFail)?;
+                .get(&Relocatable::from((segment_index as isize, i)))
+            {
+                Some(val) => match val.as_ref() {
+                    MaybeRelocatable::Int(num) => format!("{}", num.to_bigint()),
```

Comment:
For a later refactor: it could be a better idea to implement `Display` for `Felt` instead.

Reply:
It is implemented, the reason for using `to_bigint()` is to change the range of the felt (from [0,P] to [-P/2, P/2})